src/MerchantCenter/MerchantCenterService.php

```diff
@@ -334,4 +340,43 @@ protected function is_mc_contact_information_setup(): bool {
 
 		return $is_setup['phone_number'] && $is_setup['address'];
 	}
+	
+	/*
+	 * Check if the taxes + shipping rate and time + free shipping settings have been saved.
+	 *
+	 * @return bool If all required settings have been provided.
+	 *
+	 * @since x.x.x
+	 */
+	protected function saved_shipping_and_tax_options(): bool {
+		$merchant_center_settings = $this->options->get( OptionsInterface::MERCHANT_CENTER, [] );
+		$target_countries         = $this->get_target_countries();
+
+		// Tax options saved if: not US (no taxes) or tax_rate has been set
+		if ( in_array( 'US', $target_countries, true ) && empty( $merchant_center_settings['tax_rate'] ) ) {
+			return false;
+		}
+
+		// Free shipping saved if: not offered, OR offered and threshold not null
+		if ( ! empty( $merchant_center_settings['offers_free_shipping'] ) && is_null( $merchant_center_settings['free_shipping_threshold'] ) ) {
+			return false;
+		}
+
+		// Shipping options saved if: 'manual' OR 'flat' + records for all countries
+		if ( 'manual' === ( $merchant_center_settings['shipping_time'] ?? 'flat' ) ) {
```

Comment:
I think it would be more clear just to check if `shipping_time` and `shipping_rate` keys are set instead of setting them to a default `flat`. Since we don't seem to be using that default value anywhere.

Reply:
Good call, I'd originally seen that they were always set to `flat`, but it looks like if the tax settings are set first, these can be `null`. Updated in 47440cd4cf186c4382d6cd5464c25f9c4dba0623